### ISSUE TITLE
luacheck didn't work as expected

### ISSUE THREAD

**Issue by vm-001** [mpeterv/luacheck#227]
I'm using luacheck 1.0.0.

```lua
-- test.lua
local err = "error"
print("a ", err ". ")
```

```
$ luacheck test.lua 

Checking test.lua                                 OK

Total: 0 warnings / 0 errors in 1 file
```

```
$ lua test.lua

lua: test.lua:2: attempt to call local 'err' (a string value)
stack traceback:
	test.lua:2: in main chunk
	[C]: ?

```

**Comment by alerque** [comment]
Two things:

1. This is not the right repository to report bugs. See #198 for the backstory, then please close this issue and if necessary open a new one from the canonical repository here: https://github.com/lunarmodules/luacheck

2. Your report doesn't explain what you expected to be different. The issue I see is a type error (you are setting `err` to a string not a function, then using it as a function). Please be explicit in your report when you open a new issue about what you think should have happened and why it should be a lint error.

Thanks.

**Comment by vm-001** [comment]
OK Cool.
